api/gateway-operator/v1beta1/watch_namespaces_types.go

```diff
@@ -0,0 +1,39 @@
+package v1beta1
+
+// WatchNamespaces defines the namespaces to watch for resources
+//
+// +kubebuilder:validation:XValidation:message="List is required when type is List", rule="self.type == 'List' ? has(self.list) : true"
+// +kubebuilder:validation:XValidation:message="List must not be specified when type is not List", rule="self.type != 'List' ? !has(self.list) : true"
+// +apireference:kgo:include
+type WatchNamespaces struct {
+	// Type indicates the type of namespace watching to be done.
+	// By default, all namespaces are watched.
+	//
+	// +required
+	Type WatchNamespacesType `json:"type"`
+
+	// List is a list of namespaces to watch for resources.
+	// Only used when Type is set to List.
+	//
+	// +optional
+	List []string `json:"list,omitempty"`
+}
+
+// WatchNamespacesType indicates the type of namespace watching to be done.
+//
+// +kubebuilder:validation:Enum=All;List;Own
+type WatchNamespacesType string
+
+const (
+	// WatchNamespacesTypeAll indicates that all namespaces should be watched
+	// for resources.
+	WatchNamespacesTypeAll WatchNamespacesType = "All"
+	// WatchNamespacesTypeList indicates that only the namespaces listed in
+	// the Namespaces field should be watched for resources.
+	// All the namespaces enumerated in the list will be watched in addition to
+	// the namespace of the object.
+	WatchNamespacesTypeList WatchNamespacesType = "List"
+	// WatchNamespacesTypeOwn indicates that only the namespace of the
+	// object should be watched for resources.
+	WatchNamespacesTypeOwn WatchNamespacesType = "Own"
+)
```

Comment:
In other places for enums we have lowercase values

Reply:
It seems we have a mix of lowercase and capitalized: https://github.com/search?q=repo%3AKong%2Fkubernetes-configuration%20kubebuilder%3Avalidation%3AEnum&type=code but I'm fine doing the lowercase. Seems to fit a bit better here.

f2d8c4f1a3e40b8a0fea3f7faf978e3e0f3c4736